### PR TITLE
Delegate event

### PR DESCRIPTION
**Pull request by alicevhenshaw**
## Description

Added an event to Delegate.provideOrder, outputted on successful swap.

Quick description:

- [X] N/A

## Tests
All passing

## Test Coverage
All passing

### CODE REVIEW

File: source/delegate/contracts/interfaces/IDelegate.sol [hunk 1]
```diff
@@ -42,6 +42,15 @@ interface IDelegate {
     address indexed signerToken
   );
 
+  event ProvideOrder(
+    address indexed ruleOwner,
```
Comment (dmosites) [inline source/delegate/contracts/interfaces/IDelegate.sol]: `delegateOwner`?

Comment (alicevhenshaw) [reply source/delegate/contracts/interfaces/IDelegate.sol]: its called `delegateOwner` in the other events currently. (setrule and unsetrule) but could rename it in those too

Comment (dmosites) [reply source/delegate/contracts/interfaces/IDelegate.sol]: I see. Maybe similar to `staker` elsewhere this could just be `owner` in both cases.

Comment (poiresel) [reply source/delegate/contracts/interfaces/IDelegate.sol]: Did you think it was worth having both owner + tradeWallet but indexing by owner?

**Pull request merged**